### PR TITLE
Batched Avro export [VS-630]

### PR DESCRIPTION
**Pull request by mcovarr**
To address scalability failings with unbatched Avro exports.

### CODE REVIEW

**Review by rsasch — APPROVED** [review]
Looks great!  Test run with AoU 194k at https://job-manager.dsde-prod.broadinstitute.org/jobs/da1e40cb-e10b-476b-9c5e-3673582795b7.

File: scripts/variantstore/wdl/GvsExtractAvroFilesForHail.wdl [hunk 1]
```diff
@@ -183,36 +187,45 @@ task ExtractFromSuperpartitionedTables {
...
         String avro_sibling
-        Int table_index
+        Int num_superpartitions
+        Int shard_index
+        Int num_shards
     }
     parameter_meta {
         avro_sibling: "Cloud path to a file that will be the sibling to the 'avro' 'directory' under which output Avro files will be written."
-        table_index: "1-based index for the superpartitioned ref_ranges and vet tables to be extracted by this call."
+        num_superpartitions: "Total number of superpartitions requiring extraact"
```
Comment (RoriCremer) [inline scripts/variantstore/wdl/GvsExtractAvroFilesForHail.wdl:196]:
```suggestion
        num_superpartitions: "Total number of superpartitions requiring extract"
```

extraaaaact

File: scripts/variantstore/wdl/GvsExtractAvroFilesForHail.wdl [hunk 2]
```diff
@@ -183,36 +187,45 @@ task ExtractFromSuperpartitionedTables {
...
+        for superpartition in $(seq ~{shard_index + 1} ~{num_shards} ~{num_superpartitions})
+        do
+            str_table_index=$(printf "%03d" $superpartition)
+
+            # These bq exports error out if there are any objects at the sibling level to where output files would be written
+            # so an extra layer of `vet_${str_table_index}` is inserted here.
+            bq query --nouse_legacy_sql --project_id=~{project_id} "
+                EXPORT DATA OPTIONS(
+                uri='${avro_prefix}/vets/vet_${str_table_index}/vet_${str_table_index}_*.avro', format='AVRO', compression='SNAPPY') AS
+                SELECT location, sample_id, ref, REPLACE(alt,',<NON_REF>','') alt, call_GT as GT, call_AD as AD, call_GQ as GQ, cast(SPLIT(call_pl,',')[OFFSET(0)] as int64) as RGQ
```
Comment (RoriCremer) [inline scripts/variantstore/wdl/GvsExtractAvroFilesForHail.wdl:216]: I dont think we need to extract PLs? it doesn't look like Tim uses them. Let's ask on Wednesday. (Fine to include for now, I just happened to notice it in this pr)

**Review by RoriCremer — APPROVED** [review]

File: scripts/variantstore/wdl/GvsExtractAvroFilesForHail.wdl [hunk 3]
```diff
@@ -24,16 +25,19 @@ workflow GvsExtractAvroFilesForHail {
             dataset = dataset
     }
 
-    # Superpartitions have max size 4000. The inner '- 1' is so the 4000th (and multiples of 4000) sample lands in the
-    # appropriate partition, the outer '+ 1' is to iterate over the correct number of partitions.
-    scatter (i in range(((CountSamples.num_samples - 1) / 4000) + 1)) {
+    Int num_samples = CountSamples.num_samples
+    Int num_superpartitions = if (num_samples % 4000 == 0) then num_samples / 4000 else (num_samples / 4000 + 1)
```
Comment (RoriCremer) [inline scripts/variantstore/wdl/GvsExtractAvroFilesForHail.wdl:29]: a comment on this math would be helpful

**Pull request merged**